### ISSUE TITLE
DEBUG ASSERT on shutdown: PlayerInfo::instance()

### ISSUE THREAD

**Issue by uklotzde** [mixxxdj/mixxx#12242]
### Bug Description

```
DEBUG ASSERT: "s_pPlayerInfo" in function static PlayerInfo& PlayerInfo::instance() at /home/uk/volumes/Build/cpp/mixxx/src/mixer/playerinfo.cpp:41
```

```
#8  0x00000000006b5985 in mixxx_debug_assert(char const*, char const*, int, char const*) (assertion=<optimized out>, file=<optimized out>, line=<optimized out>, function=<optimized out>)
    at /home/uk/volumes/Build/cpp/mixxx/src/util/assert.h:9
#9  0x0000000000bfaa25 in mixxx_debug_assert_return_true(char const*, char const*, int, char const*)
    (function=0x602bd0 "static PlayerInfo& PlayerInfo::instance()", line=41, file=0x66fa90 "/home/uk/volumes/Build/cpp/mixxx/src/mixer/playerinfo.cpp", assertion=0x649b2c "s_pPlayerInfo")
    at /home/uk/volumes/Build/cpp/mixxx/src/util/assert.h:18
#10 PlayerInfo::instance() () at /home/uk/volumes/Build/cpp/mixxx/src/mixer/playerinfo.cpp:41
#11 0x0000000000967d44 in EngineRecord::metaDataHasChanged() (this=this@entry=0x213d8b0) at /home/uk/volumes/Build/cpp/mixxx/src/engine/sidechain/enginerecord.cpp:78
#12 0x000000000096b6a8 in EngineRecord::process(float const*, int) (this=0x213d8b0, pBuffer=0x7f33fff00040, iBufferSize=36864)
    at /home/uk/volumes/Build/cpp/mixxx/src/engine/sidechain/enginerecord.cpp:207
#13 0x000000000096e3d2 in EngineSideChain::run() (this=0x1de12f0) at /home/uk/volumes/Build/cpp/mixxx/src/engine/sidechain/enginesidechain.cpp:117
```

### Version

main + 2.4 / Qt5

### OS

Fedora 38

**Comment by uklotzde** [comment]
Only happened once yet.

**Comment by uklotzde** [comment]
Unlikely that this will ever get resolved.

**Comment by daschuer** [comment]
That is you opinion and no reason to close this bug, scheduled for 2.4.0.

**Comment by uklotzde** [comment]
No one other than me ever experienced this bug. I have opened this issue and I am free to close it whenever I decide to do so. Please respect my decision. Thank you.

**Comment by ywwg** [comment]
@daschuer I agree with @uklotzde here, let's close this because we can't reproduce it. If we do, *then* we can reopen.  I don't want to get into an open/close fight so please let me know if you agree.

**Comment by daschuer** [comment]
From the backtrace is pretty clear what happens. The fix is here #12341
